retrier/retrier_test.go

```diff
@@ -153,6 +153,24 @@ func TestRetrierRunFnWithInfinite(t *testing.T) {
 	}
 }
 
+func TestRetrierWithDynamicBackoff(t *testing.T) {
+	r := New([]time.Duration{0, 10 * time.Millisecond}, nil)
+	st := time.Now()
+
+	err := r.Run(genWork([]error{ErrWithBackoff(errFoo, 500*time.Millisecond)}))
```

Comment:
Prefer to swap these times (so the retrier defaults to 500ms and the error overrides it to 10ms), so the test runs that much faster.